src/main/java/taxi/web/filter/AuthenticationFilter.java

```
@@ -0,0 +1,56 @@
+package taxi.web.filter;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import taxi.lib.Injector;
+import taxi.service.DriverService;
+
+public class AuthenticationFilter implements Filter {
+    private static final Injector injector = Injector.getInstance("taxi");
+    private final DriverService driverService =
+            (DriverService) injector.getInstance(DriverService.class);
+    private Set<String> allowedUrls;
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        Filter.super.init(filterConfig);
+        allowedUrls = new HashSet<>();
+        allowedUrls.add("/login");
+        allowedUrls.add("/drivers/add");
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse resp = (HttpServletResponse) response;
+        HttpSession session = req.getSession();
+        Long userId = (Long) session.getAttribute("user_id");
+        if (userId == null && allowedUrls.contains(req.getServletPath())) {
+            chain.doFilter(req, resp);
+            return;
+        }
+        try {
+            if (userId == null) {
+                throw new NoSuchElementException();
+            }
+            driverService.get(userId);
+        } catch (NoSuchElementException e) {
+            session.invalidate();
+            resp.sendRedirect("/login");
+            return;
+        }
```

don't need to use try catch here and throw exception and invalidate session